Twig 3 support

xvilo opened the pull request:
- Add twig 3 as a dependency
- Update the stub environment to reflect class changes
- Fixes Code style issues that
- Fixes PHP Unit warnings, so all green now :) 
- Change php required version 7.2 (as 7.1 is not supported by twig anymore)
    - Update the travis file to include 7.2 -> 7.4 versions which are currently maintained 
- Should probably be a new major version (2.0.0)

xvilo commented:
Tests are now passing now, fixed some stuff and added Twig 3 as dep. Can this be reviewed @sserbin?

xvilo commented:
Hi @sserbin, when do you expect to take a look at this? We would love to start using twig 3 with your linter :)

sserbin commented on src/StubEnvironment.php:
```diff
@@ -13,7 +14,7 @@ class StubEnvironment extends Environment
      * {@inheritdoc}
      */
-    public function getFilter($name)
+    public function getFilter(string $name): ?TwigFilter
```
if I'm not mistaken, a signature like below would be compatible with both 2.x and 3.x of twig
```php
public function getFilter($name): ?TwigFilter
```

xvilo replied:
Yes, this worked for all but `getTest`, I have added an if statement with a comment to clarify.

sserbin commented on composer.json:
```diff
@@ -16,10 +16,10 @@
         "symfony/console": "^4.1",
         "symfony/finder": "^4.1",
-        "twig/twig": "^2.5",
+        "twig/twig": "3",
```
this could be made compatible for both with `^2.5 || ^3`

xvilo commented:
@sserbin I have added twig 2 compatibilities again
1. update composer file
2. changed inline types to only include return types
3. added a small if statement for type check to make it work on both (latest minor) versions.

See:

### Twig 2
<img width="352" alt="Screenshot 2019-12-23 at 23 26 54" src="https://user-images.githubusercontent.com/390769/71383383-5fe38300-25dc-11ea-85b3-7a3fb0fa456f.png">

### Twig 3
<img width="476" alt="Screenshot 2019-12-23 at 23 25 59" src="https://user-images.githubusercontent.com/390769/71383387-6b36ae80-25dc-11ea-80bd-442b1e51fe07.png">

xvilo commented:
@sserbin what to do with the linting issues?

sserbin commented:
> @sserbin what to do with the linting issues?

fixed those ;)
